cookbooks/child-workflows/child-workflows-megafood.md

```diff
@@ -0,0 +1,321 @@
+2. Synchronously - This will block like a regular function call and is invoked as such. The parent workflow won't continue until the child workflow has completed.
+
+### Child workflow limitations
+
+Since workflows are the main building blocks of Cadence, there are very few limitations to child workflows. 
+
+owever one to be aware of is that there is no shared state between the parent and child workflows. Asynchronous communication can be built between the workflow instances, but if there is constant communication of state going back and forth, its probably a better candidate for a single workflow..
+
+## Use Case Example: Instafood meets MegaBurgers
+
+In order to see this pattern in action, we'll create some child workflows in our sample project.
+
+### Instafood Brief
+
+Instafood is an online app-based meal delivery service. Customers can place an order for food from their favorite local restaurants via Instafood’s mobile app. Orders can be for pickup or delivery. If delivery is chosen, Instafood will organize to have one of their many delivery drivers pickup the order from the restaurant and deliver it to the customer. Instafood provides each restaurant a kiosk/tablet which is used for communication between Instafood and the restaurant. Instafood notifies the restaurant when an order is placed, and then the restaurant can accept the order, provide an ETA, mark it as ready, etc. For delivery orders, Instafood will coordinate to have a delivery driver pick up based on the ETA.
+
+### Ordering from "MegaBurgers"
+
+MegaBurgers is a large multinational fast food hamburger chain. They have an existing mobile app and website that uses a back-end REST API for customers to place orders. Instafood’s backend will directly integrate with MegaBurgers’s existing REST-based ordering system to place orders and receive updates.
+
+![MegaBurger's order status](images/Diagram_3.png)
+
+*Fig 3. MegaBurger's order state machine*
+
+We are designing Instafood to offer meals from any restaurant that signs up to our service. Each company will have its own method to take orders and monitor progress, which makes it a perfect candidate to implement via a child workflow.
+
+In the case of MegaBurger this is done via the following **if** statment, which is made at the time of ordering.
+
+``` java
+  if (Restaurant.MEGABURGER.equals(order.getRestaurant())) {
+      MegaBurgerOrderWorkflow megaBurgerOrderWorkflow = Workflow.newChildWorkflowStub(MegaBurgerOrderWorkflow.class);
+      Async.procedure(megaBurgerOrderWorkflow::orderFood, order);
+  }
+```
+The above example shows how simple this makes the parent workflow, it doesn't have to know about the various peculiarities of each company's order process, thats the job of the child workflow to implement.
```

Comment:
"it doesn't have to know about the various peculiarities of each company's order process, thats the job of the child workflow to implement." -> "it doesn't have to know about the various peculiarities of each company's order process. That's the job of the child workflow to implement."